azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -126,30 +134,49 @@ public Template getTemplate() {
    public DeploymentBody getDeploymentTemplate() {
 
       addStorageResource();
-      addVirtualNetworkResource();
       addPublicIpAddress();
+      addNetworkSecurityGroup();
       addNetworkInterfaceCard();
       addVirtualMachine();
 
+
+      DeploymentTemplate.TemplateParameters templateParameters = DeploymentTemplate.TemplateParameters.create(null);
+      DeploymentTemplate.Parameters parameters = DeploymentTemplate.Parameters.create(null);
```

Comment:
Initialize in an else clause to avoid creating an object just to immediately override it?